cmake/modules/RootMacros.cmake

```diff
@@ -564,7 +564,10 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
     foreach(dep ${ARG_DEPENDENCIES})
       set(dependent_pcm ${libprefix}${dep}_rdict.pcm)
```

Comment:
In order to better test this I think essential piece of code, I'd propose to remove this condition here, remove the `if (runtime_cxxmodules)` branch and set all target property to `ARG_NO_CXXMODULE` if runtime_cxxmodules is off.

Reply:
I re-pushed the changes taking this in mind.